_rules/css-orientation-lock-media-queries-b33eff.md

```diff
@@ -0,0 +1,198 @@
+---
+id: b33eff
+name: Page has no orientation lock specified using CSS media queries
+rule_type: atomic
+description: |
+  This rule checks that page content is not locked to any specific display orientation using CSS media queries, and the content is operable in all display orientations.
+accessibility_requirements:
+ wcag20:1.3.4: # Orientation
+  forConformance: true
+  failed: not satisfied
+  passed: further testing needed
+  inapplicable: further testing needed
+input_aspects:
+- DOM Tree
+- CSS Styling
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any `element` that is [visible](#visible) and has CSS media query styles targeting the display orientation.
+
+## Expectation
+
+Each target element does not restrict its view and operation to a single display orientation, unless a specific display orientation is [essential](https://www.w3.org/TR/WCAG21/#dfn-essential).
+
+**Note:** Examples where a particular display orientation may be [essential](https://www.w3.org/TR/WCAG21/#dfn-essential) are a bank check, a piano application, slides for a projector or television, or virtual reality content where binary display orientation is not applicable.
+
+## Assumptions
+
+- The rule does not consider the existence of any control on the page that can change the orientation on demand.
+
+## Accessibility Support
+
+*There are no major accessibility support issues known for this rule.*
+
+## Background
+
+- [Understanding Success Criterion 1.3.4: Orientation](https://www.w3.org/WAI/WCAG21/Understanding/orientation.html)
+- [Managing screen orientation](https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model/Managing_screen_orientation)
+- [MDN Orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/orientation)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+A page, where CSS media query styles negate orientation lock, and therefore does not restrict its view or operation.
+
+```html
+<html lang="en">
+  <head>
+    <!--
+      1. The style from the relative stylesheet applies `transform: rotate(90deg)`
+    -->
+    <link rel="stylesheet" href="../test-assets/css-orientation-lock-media-queries-b33eff/html-css-lock.css">
+    <!--
+      2. The below style negates the orientation lock
+    -->
+    <style>
+      @media (orientation: portrait) {
+        body {
+          transform: rotate(-90deg);
+        }
+      }
+    </style>
+  </head>
+  <body>
+    <main>
+      Page Content
+    </main>
+  </body>
+</html>
+```
+
+#### Passed example 2
+
+A page (piano application), where orientation lock is essential.
+
+```html
+<html lang="en">
+  <head>
+    <style>
+      @media (orientation: portrait) {
+        body {
+          transform: rotate(90deg);
+        }
+      }
+    </style>
+  </head>
+  <body>
+    <main>
+      <!-- piano application -->
```

Comment:
I hate to do this, because of the implications of this. But this isn't a proper test. Test cases have to actually pass or fail the rule. We need an actual test case of the orientation being essential, otherwise this rule is invalid.

That's kind of my concern with including the exception in the rule at all. We don't have any examples of it.

Reply:
We can create a simple example by having an image that is quite wide (wider than the height of most screens) and then simply stating in the text that for the full effect of the artwork to work, you should view it all at once. That should make the orientation essential.

Reply:
@WilcoFiers you are right. I cannot seem to come up with a passed example where we can emphasise that the orientation is essential. But having removed this example & the not in expectation, I believe this rule conveys that orientation if provided and discerned as lock will be a failure, in other cases we either flag as inapplicable or pass accordingly.

Reply:
@ShadowBB even having a large image as you suggested does not solve for `We need an actual test case of the orientation being essential`.

Reply:
> 
> 
> @ShadowBB even having a large image as you suggested does not solve for `We need an actual test case of the orientation being essential`.

I still would say that is possible by simply stating that the artist of the images intended it to be viewed all at once. Even if the image isn't particularly better if viewed all at once, the statement would make the orientation essential for "being able to verify that claim".